File: A3/A3-009/A3-009.py
# Input amount of customer of pods
N, K = map(int, input().split())

# Input queue line of each customer
total_customer = 0
has_customer = 0
queue = [0] * K
for _ in range(N):
    # Input queue line
    line = int(input()) - 1
    # Update total customer
    total_customer += 1
    # Update queue line amount that has customer
    if(queue[line] == 0):
        has_customer += 1
    # Assign a customer to a queue
    queue[line] += 1
    
    # Check if the customer are ready to get into a pod
    if(has_customer == K):
        # Update all queue lines
        for i in range(K):
            queue[i] -= 1       # Remove a customer from a queue
            total_customer -= 1 # Update total customer
            # Update queue line amount that has customer
            if(queue[i] == 0):
                has_customer -= 1
# Output
print(total_customer)

File: A3/A3-009/A3-009.cpp
#include<iostream>
using namespace std;

int N, K, queue[300], hasCustomer = 0, totalCustomer = 0;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);
    // Input amount of customer and pods
    cin >> N >> K;
    // Input queue line of each customer
    while(N--)
    {
        // Input queue line
        int line;
        cin >> line;
        // Update total customer
        totalCustomer++;
        // Update queue line amount that has customer
        if(queue[line - 1] == 0) { hasCustomer++; }
        // Assign a customer to a queue
        queue[line - 1]++;

        // Check if the customers are ready to get into a pod
        if(hasCustomer == K)
        {
            // Update all queue lines
            for(int i = 0; i < K; i++)
            {
                queue[i]--;         // Remove a customer from a queue
                totalCustomer--;    // Update total customer
                // Update queue line amount that has customer
                if(queue[i] == 0) { hasCustomer--; }
            }
        }
    }

    // Output
    cout << totalCustomer << '\n';
    return 0;
}